prov/sockets/src/sock_epoll.c

```diff
@@ -108,10 +112,14 @@ void sock_epoll_close(struct sock_epoll_set *set)
 
 int sock_epoll_create(struct sock_epoll_set *set, int size)
 {
+	int ret;
 	set->size = size;
 	set->used = 0;
 	set->pollfds = calloc(size, sizeof(struct pollfd));
-	return set->pollfds ? 0 : -1;
+	ret = set->pollfds ? 0 : -1;
+	if (ret < 0)
+		free(set->pollfds);
```

Comment:
Yes, we are freeing set->pollfds if calloc fails. The calling code reports error if this function returns -1. So return 0 for calloc success and -1 for fail.